crates/matrix-sdk/src/room/common.rs

```diff
@@ -931,6 +934,64 @@ impl Common {
             Err(Error::NoOlmMachine)
         }
     }
+
+    /// Get a permalink to this room.
+    ///
+    /// If this room has an alias, we use it. Otherwise, we try to use the
+    /// synced members in the room for [routing] the room ID.
+    ///
+    /// This currently returns a `matrix.to` URI but the format of the permalink
+    /// might change without notice so don't rely on it.
+    ///
+    /// [routing]: https://spec.matrix.org/v1.3/appendices/#routing
+    pub async fn permalink(&self) -> Result<String> {
```

Comment:
Why have a generic method like `permalink` instead of letting the user choose what they want (`https://matrix.to` or `matrix://`) by having separate methods for both, like in Ruma?

Reply:
I see it more as a method to get an opinionated permalink that is widely supported accross the community.

If you'd prefer a method where the permalink's format is chosen by the user, I can do that instead.

Reply:
What do you mean by the raw string without prefix? The format of the identifier is different between `matrix.to` and `matrix:` URIs.